src/commands/utils/uploadUDF.test.ts

```diff
@@ -22,120 +39,259 @@ describe("uploadUDF utils", () => {
       api_version: PresignedUploadUrlArtifactV1PresignedUrl200ResponseApiVersionEnum.ArtifactV1,
       kind: PresignedUploadUrlArtifactV1PresignedUrl200ResponseKindEnum.PresignedUrl,
+      upload_url: "https://example.com/presigned-url",
     };
+    let tempJarPath: string;
+    let tempJarUri: vscode.Uri;
+    let uploadFileToAzureStub: sinon.SinonStub;
 
-    sandbox.stub(sidecar, "getSidecar").resolves({
-      getFlinkPresignedUrlsApi: () => ({
-        presignedUploadUrlArtifactV1PresignedUrl: sandbox.stub().resolves(mockResponse),
-      }),
-    } as any);
-    const showErrorStub = sandbox.stub(notifications, "showErrorNotificationWithButtons");
+    beforeEach(() => {
+      const tempDir = os.tmpdir();
+      tempJarPath = path.join(tempDir, `test-udf-${Date.now()}.jar`);
+      fs.writeFileSync(tempJarPath, "dummy jar content");
+      tempJarUri = vscode.Uri.file(tempJarPath);
 
-    await handlePresignedUrlRequest(mockRequest);
+      // Simple mock Response - follows the pattern from uploadToAzure.test.ts
+      const mockResponse = new Response(null, { status: 200, statusText: "OK" });
+      uploadFileToAzureStub = sandbox
+        .stub(uploadToAzure, "uploadFileToAzure")
+        .resolves(mockResponse);
+    });
+    afterEach(() => {
+      // Clean up the temporary file after each test
+      if (fs.existsSync(tempJarPath)) {
+        fs.unlinkSync(tempJarPath);
+      }
+    });
+    it("should log the message confirming the upload", async () => {
+      const mockParams = {
+        environment: "env-123456",
+        cloud: "Azure",
+        region: "australiaeast",
+        artifactName: "test-artifact",
+        fileFormat: "jar",
+        selectedFile: tempJarUri,
+      };
+
+      const mockProgress = {
+        report: sandbox.stub(),
+      };
+
+      const mockToken = {
+        isCancellationRequested: false,
+        onCancellationRequested: sandbox.stub(),
+      };
+
+      const withProgressStub = sandbox.stub(vscode.window, "withProgress");
+      withProgressStub.callsFake(async (options, callback) => {
+        return await callback(mockProgress as any, mockToken as any);
+      });
+
+      await handleUploadFile(mockParams, mockPresignedUrlResponse);
+      sinon.assert.calledOnce(uploadFileToAzureStub);
+      sinon.assert.calledWith(uploadFileToAzureStub, {
+        file: sinon.match.any, // The blob object
+        presignedUrl: mockPresignedUrlResponse.upload_url,
+        contentType: "application/java-archive",
+      });
 
-    sinon.assert.calledOnceWithExactly(
-      showErrorStub,
-      "Failed to get presigned upload URL. See logs for details.",
-    );
+      sinon.assert.calledWith(mockProgress.report, { message: "Preparing file..." });
+      sinon.assert.calledWith(mockProgress.report, { message: "Uploading to Azure storage..." });
+    });
+
+    it("should handle upload errors properly", async () => {
+      const mockParams = {
```

Comment:
`mockParams` not moved out to `describe` here because it depends on `tempJarUri` which would not be initialized at that point.